### PR TITLE
[feat] address management

### PR DESCRIPTION
**Pull request by Hamdrive**
## Overview

Successfully implemented address management feature.

Features:
1. User can add/edit/delete addresses from their profile page
2. User can select between multiple addresses during checkout
3. User cannot place order without selecting an address (disabled)
4. Order summary contains delivery address

## Previews

image
image
image

### CODE REVIEW

File: src/components/AddressModal.jsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+  fillAddress,
+  setFillAddress,
+  updateAddress,
+  newAddress,
+}) => {
+  const [address, setAddress] = useState(initialAddressState);
+  console.log(fillAddress);
+
+  console.log(address);
+
```
Comment (prachi-sahani) [inline src/components/AddressModal.jsx]: can be removed

File: src/components/AddressModal.jsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    const addId = { ...address, _id: uuid() };
+    setAddress(addId);
+    newAddress(addId);
+    setAddress(initialAddressState);
+    closeModal();
+  };
+
+  const handleUpdateSubmit = (e) => {
+    e.preventDefault();
+    console.log(address)
```
Comment (prachi-sahani) [inline src/components/AddressModal.jsx]: can be removed

**Review by prachi-sahani — COMMENTED** [review]
implementation of address management looks fine
clean UI
Good to merge

File: src/components/AddressModal.jsx (hunk 1 above)
Comment (Hamdrive) [reply src/components/AddressModal.jsx]: Done

File: src/components/AddressModal.jsx (hunk 2 above)
Comment (Hamdrive) [reply src/components/AddressModal.jsx]: Done

**Review by shivampandey0 — APPROVED** [review]
Good use of custom form validation with CustomHook, works as expected.

Good to merge

**Pull request merged**